src/pages/workspace/workflows/WorkspaceWorkflowsPayerPage.tsx

```diff
@@ -0,0 +1,238 @@
+                    textStyles={styles.textStrong}
+                    badgeStyles={[styles.justifyContentCenter, StyleUtils.getMinimumWidth(60), styles.badgeBordered]}
+                />
+            );
+
+            const isAuthorizedPayer = policy?.reimburserEmail === details?.login ?? policy?.reimburserAccountID === accountID;
+
+            const formattedMember = {
+                keyForList: accountIDKey,
+                accountID,
+                isSelected: isAuthorizedPayer,
+                isDisabled: policyMember.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE || !isEmptyObject(policyMember.errors),
+                text: formatPhoneNumber(PersonalDetailsUtils.getDisplayNameOrDefault(details)),
+                alternateText: formatPhoneNumber(details?.login ?? ''),
+                rightElement: roleBadge,
+                icons: [
+                    {
+                        source: UserUtils.getAvatar(details?.avatar, accountID),
+                        name: formatPhoneNumber(details?.login ?? ''),
+                        type: CONST.ICON_TYPE_AVATAR,
+                        id: accountID,
+                    },
+                ],
+                errors: policyMember.errors,
+                pendingAction: policyMember.pendingAction ?? isAuthorizedPayer ? policy?.pendingFields?.reimburserEmail : null,
+            };
+
+            if (policy?.reimburserEmail === details?.login ?? policy?.reimburserAccountID === accountID) {
+                authorizedPayerDetails.push(formattedMember);
+            } else {
+                policyAdminDetails.push(formattedMember);
+            }
+        });
+        return [policyAdminDetails, authorizedPayerDetails];
+    }, [
+        personalDetails,
+        policyMembers,
+        translate,
+        policy?.reimburserEmail,
+        isDeletedPolicyMember,
+        policy?.owner,
+        styles,
+        StyleUtils,
+        policy?.reimburserAccountID,
+        policy?.pendingFields?.reimburserEmail,
+    ]);
+
+    const sections: MembersSection[] = useMemo(() => {
+        const sectionsArray: MembersSection[] = [];
+
+        if (searchTerm !== '') {
+            const filteredOptions = [...formattedPolicyAdmins, ...formattedAuthorizedPayer].filter((option) => {
+                const searchValue = OptionsListUtils.getSearchValueForPhoneOrEmail(searchTerm);
+                return !!option.text?.toLowerCase().includes(searchValue) || !!option.login?.toLowerCase().includes(searchValue);
+            });
+            return [
+                {
+                    title: undefined,
+                    data: filteredOptions,
+                    shouldShow: true,
+                },
+            ];
+        }
+
+        sectionsArray.push({
+            data: formattedAuthorizedPayer,
+            shouldShow: true,
+            indexOffset: formattedPolicyAdmins.length,
```

Comment:
Getting the index incorrect here caused the issue in https://github.com/Expensify/App/issues/38203.